src/odemis/acq/align/test/autofocus_test.py

```diff
@@ -602,5 +603,76 @@ def test_multi_det(self):
         self.assertEqual(len(res), ngs * nds)
 
 
+class TestAutofocus1d(unittest.TestCase):
+    """
+    Test autofocus functions on 1 line CCD.
+    """
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            test.start_backend(SPARC2_FOCUS_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # find components by their role
+        cls.ccd = model.getComponent(role="ccd")
+        cls.spectrometer = model.getComponent(role="spectrometer-integrated")
```

Comment:
If removing the special code to prevent hardware binning, it might work less well. Please check whether the test cases pass (probably), or you need to introduce extra tricks to simulate a 1 line CCD properly.